app-editors/vis/vis-9999.ebuild

```diff
@@ -2,25 +2,27 @@
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
-inherit git-r3
+LUA_COMPAT=( lua5-2 lua5-3 )
+
+inherit lua-single git-r3
 
 DESCRIPTION="modern, legacy free, simple yet efficient vim-like editor"
 HOMEPAGE="https://github.com/martanne/vis"
 EGIT_REPO_URI="https://github.com/martanne/vis.git"
 LICENSE="ISC"
 SLOT="0"
 KEYWORDS=""
-IUSE="+ncurses selinux test tre"
+IUSE="+ncurses +lua +lpeg selinux test tre"
 RESTRICT="!test? ( test )"
 
 # - Known to also work with NetBSD curses
-# - ::lua package done for using >=dev-lang/lua-5.2
-# which is needed for syntax highlighting and settings but masked in ::gentoo
 DEPEND="dev-libs/libtermkey
 	ncurses? ( sys-libs/ncurses:0= )
+	lua? ( ${LUA_DEPS} )
 	tre? ( dev-libs/tre:= )"
 RDEPEND="${DEPEND}
-	app-eselect/eselect-vi"
+	app-eselect/eselect-vi
+	lua? ( lpeg? ( >=dev-lua/lpeg-0.12 ) )"
```

Comment:
Hmm, this probably needs to be wrapped inside `lua_gen_cond_dep` but not sure whether it works, at all, before it's migrated to use new lua eclasses. Thoughts?

Reply:
1a6dc03ce8e11606e794ea6d187771e18432153c it's done now.